### PR TITLE
Fix issue #67 - Create event card

### LINKED ISSUE #67: Create event card
Create an event card based on the figma design (check the wiki)

DoD:
- A card component that accepts date, title, time, description, and address
- The show on map just needs to be a button, no functionality should be implemented
- The card should fill the width of its container
- Uses the proper text sizing, button components
- Below is a given sample for testing with

Header: Git Workshop with Linus Torvalds
Time: 12:00 PM
Description: Want to get better at git? Want to learn how you can contribute to open source? Learn from the greatest! Linus Torvalds! This workshop will cover forking repos, guidelines, pull requests, and why you should contribute! Food will be provided immediately after as well as mercy. Merchandise will also be available! This event will be hosted in the CS Lounge!
Date: 01/01/2022

### PR DESCRIPTION
**Pull request by ThatMarko**
Provide two events with default dummy data. This successfuly resolves #67. Please review, need this pull request to continue working on #69.

### CODE REVIEW

File: src/pages/eventdetails.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
-import './eventdetails.css'
+import EventCard from "../components/EventCard";
+import "./eventdetails.css";
 
-export default function EventDetails(){
-    return (<h1>Event Details</h1>)
+const dummyData = {
```
Comment (ChristianBingman) [inline src/pages/eventdetails.js:4]: Can this be moved into a separate data file for consistency and to keep this file clean?

File: src/components/PrimaryButton/index.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 export default function PrimaryButton(props) {
-    return (<button className='rounded-[40px] bg-primary-light text-white font-bold hover:bg-primary-dark py-2 px-6 w-fit'>
-              <h3>{props.innerText}</h3>
-            </button>)
-  };
\ No newline at end of file
+  return (
+    <button
+      className="rounded-[40px] bg-primary-light text-white font-bold hover:bg-primary-dark py-2 px-6 w-fit"
+      style={{ maxHeight: "75px" }}
```
Comment (ChristianBingman) [inline src/components/PrimaryButton/index.js:5]: I am confused as to what this actually does? I removed it and it doesn't change the button?

File: src/components/EventCard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <h2 className="flex-auto ml-4">{event.Header}</h2>
+        <h4 className="flex-initial justify-self-end">{event.Date}</h4>
+      </div>
+      <div className="flex">
+        <p>{event.Description}</p>
+        <div className="flex-none self-end justify-self-end">
+          <PrimaryButton innerText={"Show on Map"} />
+        </div>
+      </div>
+      <hr className="my-3" />
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:17]: The hr should be handled in the eventdetails since we don't want it on the last element.

File: src/components/EventCard/index.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export default function EventCard({ event }) {
+  return (
+    <div className="flex-row w-full">
+      <div className="flex w-full">
+        <h4 className="flex-initial">{event.Time}</h4>
+        <h2 className="flex-auto ml-4">{event.Header}</h2>
+        <h4 className="flex-initial justify-self-end">{event.Date}</h4>
+      </div>
+      <div className="flex">
+        <p>{event.Description}</p>
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:12]: There should be some sort of spacer in between these, so they don't rub against each other

File: src/components/EventCard/index.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import PrimaryButton from "../PrimaryButton";
+
+export default function EventCard({ event }) {
+  return (
+    <div className="flex-row w-full">
+      <div className="flex w-full">
+        <h4 className="flex-initial">{event.Time}</h4>
+        <h2 className="flex-auto ml-4">{event.Header}</h2>
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:8]: Could you add a bit more margin on the left?

File: src/components/EventCard/index.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import PrimaryButton from "../PrimaryButton";
+
+export default function EventCard({ event }) {
+  return (
+    <div className="flex-row w-full">
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:5]: flex-row is redundant here

File: src/components/EventCard/index.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      <div className="flex w-full">
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:6]: These items should be vertically center aligned.

File: src/components/EventCard/index.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        <h4 className="flex-initial">{event.Time}</h4>
+        <h2 className="flex-auto ml-4">{event.Header}</h2>
+        <h4 className="flex-initial justify-self-end">{event.Date}</h4>
+      </div>
+      <div className="flex">
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:11]: These items should be vertically center aligned.

File: src/components/EventCard/index.jsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+        <p>{event.Description}</p>
+        <div className="flex-none self-end justify-self-end">
```
Comment (ChristianBingman) [inline src/components/EventCard/index.jsx:13]: self-end is redundant here (and actually breaks vertical alignment)

File: src/components/EventCard/index.jsx (hunk 7 above)
Comment (ChristianBingman) [reply src/components/EventCard/index.jsx:6]: hint: (items-center)

**Review by ChristianBingman — CHANGES_REQUESTED** [review]
Sorry for the amount of changes, but I appreciate the hard work, and all of these should only take a couple minutes. I am aware of the button shadow and h4 weight being a little heavy, but don't worry about those for now.

File: src/pages/eventdetails.js (hunk 1 above)
Comment (ThatMarko) [reply src/pages/eventdetails.js]: See commit [6b3dc41](https://github.com/acm-uic/flourish-2023/pull/75/commits/6b3dc41a95f404d828a750c8ef02f51d0298a298)

File: src/components/PrimaryButton/index.js (hunk 2 above)
Comment (ThatMarko) [reply src/components/PrimaryButton/index.js:5]: I had some issues without the maxHeight styling. The button was stretched out vertically.

**Review by ChristianBingman — CHANGES_REQUESTED** [review]
It seems like a lot more than requested was modified, the dummy data should just be one element to show that it is working. A lot of the modifications you made can be cut down. There are also a couple misspellings (items-centers?) that need to be double checked. Thanks!

**Comment by ThatMarko** [comment]
TLDR; I already started working on the other issue I assigned myself to. Can we get a dummy API or something so that I can test with more data? I'm removing some of the dummy data I put into, but I will be adding it back when I start working on the other issue.

File: src/pages/eventdetails.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ export default function EventDetails() {
       <div className={"ml-[10rem] min-w-fit"}>
         <EventCard event={DummyEvents["Monday"][0]} />
         <hr className="my-3" />
-        <EventCard event={DummyEvents["Monday"][1]} />
+        <EventCard event={DummyEvents["Monday"][0]} />
       </div>
     </div>
   );
```
Comment (ChristianBingman) [inline src/pages/eventdetails.js:18]: This will all need to be rewritten anyways, but this is fine for now.

**Review by ChristianBingman — APPROVED** [review]
Looks good
